interfaces/tempo_cluster/v0/interface_tests/test_provider.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2024 Canonical
+# See LICENSE file for licensing details.
+import json
+
+from interface_tester.interface_test import Tester
+from scenario import State, Relation
+
+
+
+def test_validation_fails_with_missing_role():
+    tester = Tester(state_in=State(
+        relations=[
+            Relation(
+                endpoint='tempo_cluster',
+                interface='tempo_cluster',
+                remote_app_name='worker',
+                remote_app_data={
+                },
+                remote_units_data={
+                    0: {
+                        "juju_topology": json.dumps({
+                            "application": "worker",
+                            "unit": "worker/0",
+                            "charm_name": "worker",
+                        }),
+                        "address": json.dumps("1.2.4.5"),
```

Comment:
This is an China Network Information Center IP. Please use an [RFC 5737 IP](https://www.rfc-editor.org/rfc/rfc5737).

Also for the cases below.

Reply:
Done, good catch!